hack/dockerfiles/test.Dockerfile

```diff
@@ -57,21 +64,30 @@ FROM cross-windows AS buildd.exe
 RUN go build -o /buildd.exe ./cmd/buildd
 
 # Copy together all binaries needed for standalone mode
-FROM alpine AS buildkit-standalone
+FROM alpine AS buildkit-buildd-standalone
 COPY --from=runc /usr/bin/runc /usr/bin/
 COPY --from=buildd-standalone /usr/bin/buildd-standalone /usr/bin/
 COPY --from=buildctl /usr/bin/buildctl /usr/bin/
 ENTRYPOINT ["buildd-standalone"]
 
 # Copy together all binaries for containerd mode
-FROM alpine AS buildkit-containerd
+FROM alpine AS buildkit-buildd-containerd
 COPY --from=buildd-containerd /usr/bin/buildd-containerd /usr/bin/
 COPY --from=buildctl /usr/bin/buildctl /usr/bin/
 COPY --from=runc /usr/bin/runc /usr/bin/
 COPY --from=containerd /go/src/github.com/containerd/containerd/bin/containerd /usr/bin/
 COPY --from=containerd /go/src/github.com/containerd/containerd/bin/containerd-shim /usr/bin/
 ENTRYPOINT ["buildd-containerd"]
 
-FROM buildkit-${BUILDKIT_TARGET}
+# Copy together all binaries for "everything" mode
+FROM alpine AS buildkit-buildd
+COPY --from=buildd /usr/bin/buildd /usr/bin/
+COPY --from=buildctl /usr/bin/buildctl /usr/bin/
+COPY --from=runc /usr/bin/runc /usr/bin/
```

Comment:
nit: this can be combined with the standalone stage(or make a common shared stage).